io_scene_godot/converters/material_node_tree.py

```diff
@@ -0,0 +1,857 @@
+FRAGMENT_OUT_BSDF.set_channel('anisotropy', FRAGMENT_OUT_ANISOTROPY)
+FRAGMENT_OUT_BSDF.set_channel('transmission', FRAGMENT_OUT_TRANSMISSION)
+FRAGMENT_OUT_BSDF.set_channel('eta', FRAGMENT_OUT_ETA)
+FRAGMENT_OUT_BSDF.set_channel('emission', FRAGMENT_OUT_EMISSION)
+FRAGMENT_OUT_BSDF.set_channel('normal', FRAGMENT_OUT_NORMAL)
+FRAGMENT_OUT_BSDF.set_channel('tangent', FRAGMENT_OUT_TANGENT)
+
+
+class MaterialScript:
+    # pylint: disable-msg=too-many-public-methods
+    # FIXME:
+    """Script for godot shader material"""
+    def __init__(self):
+        self.global_codes = Array(
+            prefix='shader_type spatial;\n',
+            seperator='\n',
+            suffix='\n'
+        )
+
+        self.fragment_codes = Array(
+            prefix='\nvoid fragment() {\n\t',
+            seperator='\n\t',
+            suffix='\n}\n'
+        )
+
+        self.vertex_codes = Array(
+            prefix='\nvoid vertex() {\n\t',
+            seperator='\n\t',
+            suffix='\n}\n'
+        )
+
+        self._set_render_mode()
+        self._predefine_variables()
+
+        # cache function names to avoid duplicated
+        # function code being added
+        self.function_name_set = set()
+        self.function_codes = Array(
+            prefix='\n',
+            seperator='\n',
+            suffix='\n'
+        )
+
+        self.variable_count = 0
+
+        # maintain a mapping from socket to already
+        # defined variable or value
+        self.socket_to_var_map = dict()
+
+        self.textures = dict()
+
+        # variable may be frequently used
+        self.inv_view_mat = None
+        self.inv_world_mat = None
+
+    def _set_render_mode(self):
+        self.global_codes.append(
+            "render_mode blend_mix,depth_draw_always,"
+            "cull_back,diffuse_burley,specular_schlick_ggx;",
+        )
+
+    def _add_function(self, function):
+        if function.name not in self.function_name_set:
+            self.function_name_set.add(function.name)
+            self.function_codes.append(function.code)
+
+    def _predefine_variables(self):
+        self.fragment_var_definition(
+            FRAGMENT_OUT_COLOR, Value('vec4', [1.0] * 4)
+        )
+        self.fragment_var_definition(
+            FRAGMENT_OUT_ANISOTROPY, Value('float', 0.0)
+        )
+        self.fragment_var_definition(
+            FRAGMENT_OUT_TRANSMISSION, Value('float', 0.0)
+        )
+        self.fragment_var_definition(
+            FRAGMENT_OUT_ETA, Value('float', 1.0)
+        )
+
+    def compile(self, export_settings, has_uv):
+        """Serialization"""
+        self.fragment_codes.extend([
+            'ALBEDO = result_color.rgb;',
+            'TRANSMISSION = vec3(1.0, 1.0, 1.0) *  result_transmission;',
+        ])
+        if has_uv:
+            self.fragment_codes.append(
+                'ANISOTROPY = result_anisotropy;'
+            )
+
+        set_alpha = 'ALPHA = result_color.a;'
+        set_reflection = (
```

Comment:
Is this reflection or refraction? The GLSL code implies refraction but the variable name is reflection.

Reply:
typo, it should be refraction